#12 feat: post entity make

Pes113 opened the pull request:
# 🔥 Pull Request


## 🔗 관련 이슈 번호

<!--- ex) 이슈 번호를 작성해주세요: Closes #11 -->
- Closes #12 #13 #14 #15 

<br>

## 📌 작업 내용

- 원격 DB와 SpringBoot 연결
- 게시판 구성에 필요한 Entity, MySQL 상 DB 생성

<br>

## ✅ 변경 사항

- domain/post/controller 폴더 및 예하 파일 PostController.java 생성
- domain/post/dto 폴더 생성
- domain/repository/PostRepository 폴더 및 예하 파일 PostRepository 생성, PostRepository 내용 미작성
- domain/service 폴더 생성 및 예하 Entity(BaseLiquor, Ingredient, Post, PostBaseLiquor, PostIngredient, UserLike 생성(Entity)
- user/service/User 내 외래키 연결을 위한 코드 추가(43 ~ 47 Line) 

<br>

## 📸 스크린샷 (선택)

![스크린샷 2025-05-29 163124](https://github.com/user-attachments/assets/a74c4894-e811-43cf-bee3-fdbac4716cfb)


<br>

## 🗨️ 코멘트

- 로컬 SpringBoot 활용하여 원격으로 DB 접속 및 엔티티 구조로 테이블 생성까지 확인했습니다.
- 원격 DB 접속하시면 DB 테이블 내용 확인 가능합니다. 혹여나 실수한 부분 있으시면 수정할테니 말씀해주세요.

Pes113 committed:
#12 feat: post entity make

mirageoasis commented on back/src/main/java/com/group4/aldalka/domain/post/service/Post.java:
```diff
@@ -0,0 +1,61 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "post_id")
+    private Long post_id;
```
camelcase로 해주세요~

eunalove commented on back/src/main/java/com/group4/aldalka/domain/post/service/PostIngredient.java:
수고 많으십니다
아래의 테이블같은 경우 다대다 관계에서 쓰이는 중간테이블이고 자체적인 id를 가지고 있습니다.
유니크 제약조건을 사용해서 (post_id, ingredient_id) 조합의 중복 삽입을 방지하면 어떨까요?

Pes113 committed:
Delete .idea directory

.idea 파일 삭제

Pes113 committed:
Update BaseLiquor.java

변수 이름 변경
private Long base_liquor_id
->
private Long baseLiquorId

Pes113 committed:
Update Ingredient.java

변수 이름 변경
private Long id;
->
private Long IngredientId

Pes113 committed:
Update Post.java

변수 이름 및 자료구조 변경
private String user_id -> userId

title          -> length = "380"
content    -> length = "2200"
recipe      -> length = "2200" 
imageUrl -> length = "2083"

Pes113 committed:
Update PostBaseLiquor.java

변수 이름 변경
private Long post_base_liquor_id
->
private Long postBaseLiquorId

Pes113 committed:
Update PostIngredient.java

변수 이름 변경
private Long post_ingredient_id
->
private Long postIngredientId;

Pes113 committed:
Update UserLike.java

변수 이름 변경
private Long like_id
->
private Long likeId

Pes113 committed:
Rename : (Column 변경)

Pes113 commented:
다들 감사합니다~
